dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api_test.cc

```diff
@@ -280,11 +268,219 @@ class HwDecoderMemoryPoolTest : public ::testing::Test {
+    for (int k = 0; k < batch_size_; k++) {
+      crop_data.mutable_tensor<float>(k)[0] = crop_w;
+      crop_data.mutable_tensor<float>(k)[1] = crop_h;
+    }
+
+    pipeline_.AddOperator(
+            OpSpec("FileReader")
+                    .AddArg("device", "cpu")
+                    .AddArg("file_root", list_root)
+                    .AddOutput("compressed_images", "cpu")
+                    .AddOutput("labels", "cpu"));
+    auto decoder_spec =
+            OpSpec("ImageDecoderCrop")
+                    .AddArg("device", "mixed")
+                    .AddArg("output_type", DALI_RGB)
+                    .AddArg("hw_decoder_load", .7f)
+                    .AddArg("crop", std::vector<float>{224.0f, 224.0f})
+                    .AddInput("compressed_images", "cpu")
+                    .AddOutput("images", "gpu");
+    pipeline_.AddOperator(decoder_spec, decoder_name_);
+
+    pipeline_.Build(outputs_);
+
+    auto node = pipeline_.GetOperatorNode(decoder_name_);
+    if (!node->op->GetDiagnostic<bool>("using_hw_decoder")) {
+      if (nvml::isHWDecoderSupported()) {
+        FAIL() << "HW Decoder exists in the system and failed to open";
+      }
+      GTEST_SKIP();
+    }
+  }
+
+  int batch_size_ = 47;
+  Pipeline pipeline_{batch_size_, 1, 0, -1, false, 2, false};
+  vector<std::pair<string, string>> outputs_ = {{"images", "gpu"}};
+  std::string decoder_name_ = "Lorem Ipsum";
+};
+
+TEST_F(HwDecoderCropUtilizationTest, UtilizationTest) {
+  this->pipeline_.RunCPU();
+  this->pipeline_.RunGPU();
+
+  auto node = this->pipeline_.GetOperatorNode(this->decoder_name_);
+  auto nsamples_hw = node->op->GetDiagnostic<int64_t>("nsamples_hw");
+  auto nsamples_cuda = node->op->GetDiagnostic<int64_t>("nsamples_cuda");
+  auto nsamples_host = node->op->GetDiagnostic<int64_t>("nsamples_host");
+  EXPECT_EQ(nsamples_hw, 35) << "HW Decoder malfunction: incorrect number of images decoded in HW";
+  EXPECT_EQ(nsamples_cuda, 12);
+  EXPECT_EQ(nsamples_host, 0)
+                << "Image decoding malfuntion: all images should've been decoded by CUDA or HW";
+}
+
+
+class HwDecoderRandomCropUtilizationTest : public ::testing::Test {
+ public:
+  void SetUp() final {
+    dali::string list_root(testing::dali_extra_path() + "/db/single/jpeg");
+
+    auto shape = uniform_list_shape(batch_size_, {2});
+    TensorList<CPUBackend> begin_data;
+    begin_data.set_type(TypeInfo::Create<float>());
+    begin_data.Resize(shape);
+    float crop_x = 0.25f, crop_y = 0.124f;
+    for (int k = 0; k < batch_size_; k++) {
+      begin_data.mutable_tensor<float>(k)[0] = crop_x;
+      begin_data.mutable_tensor<float>(k)[1] = crop_y;
+    }
+
+    TensorList<CPUBackend> crop_data;
+    float crop_w = 0.5f, crop_h = 0.25f;
+    crop_data.set_type(TypeInfo::Create<float>());
+    crop_data.Resize(shape);
+    for (int k = 0; k < batch_size_; k++) {
+      crop_data.mutable_tensor<float>(k)[0] = crop_w;
+      crop_data.mutable_tensor<float>(k)[1] = crop_h;
+    }
+
```

Comment:
```suggestion
    auto shape = uniform_list_shape(batch_size_, {2});
    TensorList<CPUBackend> begin_data;
    begin_data.set_type(TypeInfo::Create<float>());
    begin_data.Resize(shape);
    float crop_x = 0.25f, crop_y = 0.124f;
    for (int k = 0; k < batch_size_; k++) {
      begin_data.mutable_tensor<float>(k)[0] = crop_x;
      begin_data.mutable_tensor<float>(k)[1] = crop_y;
    }

    TensorList<CPUBackend> crop_data;
    float crop_w = 0.5f, crop_h = 0.25f;
    crop_data.set_type(TypeInfo::Create<float>());
    crop_data.Resize(shape);
    for (int k = 0; k < batch_size_; k++) {
      crop_data.mutable_tensor<float>(k)[0] = crop_w;
      crop_data.mutable_tensor<float>(k)[1] = crop_h;
    }

```
Same here, this is not necessary for RandomCrop API

Reply:
Done